src/ml_tooling/plots/target_feature_distribution.py

```diff
@@ -0,0 +1,72 @@
+from matplotlib import pyplot as plt
+from matplotlib.axes import Axes
+import pandas as pd
+import numpy as np
+
+from ml_tooling.utils import DataType
+from scipy import stats
+
+
+def target_feature_distribution(
+    data: pd.DataFrame,
+    targets: DataType,
+    features: DataType,
+    title: str = None,
```

Comment:
For the plot functions, we want to avoid dynamic titles  - this is the lower-level plumbing if someone needs to do something custom. Targets would not have a name if it's a numpy array.

I would set the title to a default string and then the dataset plotting method will handle dynamic stuff, since we are more in control there.